### PR TITLE
Add documentation for k Shortest Paths

### PR DESCRIPTION
**Pull request by markuspf**
**Internal Referece:**
- https://github.com/arangodb/planning/issues/3179

### CODE REVIEW

**Comment by markuspf** [comment]
Still waiting for feedback on the picture of the example graph.

**Comment by goedderz** [comment]
Could you also add a paragraph to the release notes, i.e. https://docs.arangodb.com/devel/Manual/ReleaseNotes/NewFeatures35.html?

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+k Shortest Paths in AQL
+====================
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]:
```suggestion
=======================
```

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+direction for each collection in your path search.
+
+Examples
+--------
+We create a graph that reflects some possible train connections in Europe and North America.
+This graph has no claim to accuracy or completeness.
+
+![train_map](train_map.png)
+
+    @startDocuBlockInline GRAPHKSP_01_create_graph
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]: TODO: generate new examples and commit them

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  startVertex TO targetVertex
+  edges1, ANY edges2, edges3
+```
+
+All collections in the list that do not specify their own direction will use the
+direction defined after *IN* (here: *OUTBOUND*). This allows to use a different
+direction for each collection in your path search.
+
+Examples
+--------
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]:
```suggestion
--------

```

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+This type of query is supposed to find the first *k* paths in order of length
+(or weight) between two given documents, *startVertex* and *targetVertex* in
+your graph.
+
+Every such path will be returned as a JSON object with three components:
+
+1. an array of the vertices on the path,
+2. an array of the edges on the path, and
+3. the weight of the path.
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]: Are the attributes called `vertices`, `edges` and `weight`? Then we could hint this here already:

1. an array of the `vertices` on the path,
2. an array of the `edges` on the path, and
3. the `weight` of the path.

```
1. an array of the `vertices` on the path,
2. an array of the `edges` on the path, and
3. the `weight` of the path.
```

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+------
+
+The syntax for k Shortest Paths queries is similar to the one for [Shortest Path](ShortestPath.md).
+You have two options: either use a named graph, or a set of edge
+collections.
+
+It is highly recommended that you use a **LIMIT** statement,
+as k Shortest Paths is an expensive operation, and on large graphs can lead to a large
+number of paths being returned, or an expensive (but unsuccessful) search for more shortest paths
+might be performed.
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]: Make this a warning (hint box)? Is it important (enough)?

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  startVertex TO targetVertex
+  GRAPH graphName
+  [OPTIONS options]
+```
+
+- `FOR`: the variable **path** which contains one path as an object containing 
+   vertices, edges, and the weight of the path.
+- `IN` `OUTBOUND|INBOUND|ANY`: defines in which direction edges are followed
+  (outgoing, incoming, or both)
+- **startVertex** `TO` **targetVertex** (both string|object): the two vertices between
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]: K_SHORTEST_PATH is not mentioned here - I think it should, even if it's only a keyword.

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    ~addIgnoreCollection("places");
+    ~addIgnoreCollection("connections");
+    var examples = require("@arangodb/graph-examples/example-graph.js");
+    var graph = examples.loadGraph("kShortestPathsGraph");
+    db.places.toArray();
+    db.connections.toArray();
+    @END_EXAMPLE_ARANGOSH_OUTPUT
+    @endDocuBlock GRAPHKSP_01_create_graph
+
+We start with a route from **Aberdeen** to **London**, and compare the outputs of SHORTEST_PATH and K_SHORTEST_PATHS with *LIMIT 1*:
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]: avoid overly long line

```suggestion
We start with a route from **Aberdeen** to **London**, and compare the outputs
of SHORTEST_PATH and K_SHORTEST_PATHS with *LIMIT 1*:
```

File: Documentation/Books/AQL/Graphs/KShortestPaths.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+(or weight) between two given documents, *startVertex* and *targetVertex* in
+your graph.
+
+Every such path will be returned as a JSON object with three components:
+
+1. an array of the vertices on the path,
+2. an array of the edges on the path, and
+3. the weight of the path.
+
+If no *weightAttribute* is given, the weight of the path is just its length.
```
Comment (Simran-B) [inline Documentation/Books/AQL/Graphs/KShortestPaths.md]: Is the weight equal to the sum of all edge weights?

**Comment by markuspf** [comment]
I addressed some of the comments; that said, I am not happy with the picture of the graph...

**Comment by markuspf** [comment]
TODO: Also make a picture of the complete graph used in the example.

File: Documentation/Books/AQL/Graphs/KShortestPaths.md (hunk 6 above)
Comment (ifcologne) [reply Documentation/Books/AQL/Graphs/KShortestPaths.md]: Please add the Limit to the definition and detail description. Yes, there is a hint in the document - but users will just focus on the definition and run into trouble... 

```
FOR path 
  IN OUTBOUND|INBOUND|ANY K_SHORTEST_PATHS
  startVertex TO targetVertex
  GRAPH graphName
  [OPTIONS options]
  LIMIT <k>
```

Comment (markuspf) [reply Documentation/Books/AQL/Graphs/KShortestPaths.md]: done.

**Comment by Simran-B** [comment]
http://jenkins01.arangodb.biz:8080/view/Documentation/job/arangodb-documentation/415/

**Pull request merged**